### PR TITLE
deepfilternet: init at 0.5.6

### PR DESCRIPTION
**Pull request by ralismark**
## Description of changes

Based on and supersedes a part of #240048. This adds the `deepfilternet` package, which contains `deep_filter_ladspa.so` that can be used by e.g. pipewire. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ralismark** [comment]
Result of `nixpkgs-review pr 276460` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>deepfilternet-ladspa</li>
  </ul>
</details>

File: pkgs/by-name/de/deepfilternet-ladspa/package.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+,
```
Comment (MikaelFangel) [inline pkgs/by-name/de/deepfilternet-ladspa/package.nix:4]:
```suggestion
```

File: pkgs/by-name/de/deepfilternet-ladspa/package.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    mkdir $out/lib/ladspa
+    mv $out/lib/libdeep_filter_ladspa.so $out/lib/ladspa/libdeep_filter_ladspa.so
+  '';
+
+  meta = {
+    description = "Noise supression using deep filtering (LADSPA plugin)";
+    homepage = "https://github.com/Rikorose/DeepFilterNet";
+    license = with lib.licenses; [ mit asl20 ];
+    maintainers = with lib.maintainers; [ ralismark ];
+    platforms = lib.platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/by-name/de/deepfilternet-ladspa/package.nix:36]:
```suggestion
```
buildRustPackage defaults to unix

**Review by jopejoe1 — APPROVED** [review]
LGTM

**Comment by felschr** [comment]
easyeffects has an integration with DeepFilterNet based on the LADSPA plugin.
Would be nice if the integration could be implemented in this PR as it should be pretty straightforward.
The package probably only needs to be added to `buildInputs` like it was done for `rznnoise` & `speexdsp`:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/audio/easyeffects/default.nix

**Comment by ralismark** [comment]
@felschr i can add it, but I'm unsure how to test if easyeffects works with the deepfilter -- it's not immediately showing up in the filter list and I'm unsure why

**Comment by felschr** [comment]
> @felschr i can add it, but I'm unsure how to test if easyeffects works with the deepfilter -- it's not immediately showing up in the filter list and I'm unsure why

It should show up in the input effects under the term "Deep Noise Remover":

image

**Comment by felschr** [comment]
Locally I'm running easyeffects 7.1.3, that seems to be the issue, this branch is still on 7.1.0.

**Comment by felschr** [comment]
I gave this a try. Rebasing on master to get easyeffects 7.1.3 and adding `deepfilternet-ladspa` to `buildInputs` & `ladspaPlugins` makes this work.

image

**Comment by ralismark** [comment]
Thanks @felschr! Have rebased and it now shows up.

I've also renamed this to just `deepfilternet` so if/when the rest of the project gets packaged up there doesn't need to be a rename.

**Comment by felschr** [comment]
Result of `nixpkgs-review pr 276460` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>deepfilternet</li>
    <li>easyeffects</li>
    <li>easyeffects.debug</li>
    <li>gnomeExtensions.easyeffects-preset-selector</li>
  </ul>
</details>

**Review by felschr — APPROVED** [review]
Using easyeffects I can use all of the exposed deepfilter functionality.

File: pkgs/by-name/de/deepfilternet/package.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  postInstall = ''
+    mkdir $out/lib/ladspa
+    mv $out/lib/libdeep_filter_ladspa.so $out/lib/ladspa/libdeep_filter_ladspa.so
+  '';
+
+  meta = {
+    description = "Noise supression using deep filtering";
+    homepage = "https://github.com/Rikorose/DeepFilterNet";
+    license = with lib.licenses; [ mit asl20 ];
+    maintainers = with lib.maintainers; [ ralismark ];
```
Comment (felschr) [inline pkgs/by-name/de/deepfilternet/package.nix:37]:
```suggestion
    maintainers = with lib.maintainers; [ ralismark ];
    changelog = "https://github.com/Rikorose/DeepFilterNet/releases/tag/${src.rev}";
```

Small improvement.

**Review by felschr — APPROVED** [review]

File: pkgs/by-name/de/deepfilternet/package.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    };
+  };
+
+  # only the ladspa plugin part has been packaged so far...
+
+  buildAndTestSubdir = "ladspa";
+
+  postInstall = ''
+    mkdir $out/lib/ladspa
+    mv $out/lib/libdeep_filter_ladspa.so $out/lib/ladspa/libdeep_filter_ladspa.so
```
Comment (SuperSandro2000) [inline pkgs/by-name/de/deepfilternet/package.nix:30]: If replacing .so with the stdenv.hostPlatform.extensions.sharedLibrary variable doesn't fix Darwin, we could just mark it broken.

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 276460` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>deepfilternet</li>
    <li>easyeffects</li>
    <li>easyeffects.debug</li>
    <li>gnomeExtensions.easyeffects-preset-selector</li>
  </ul>
</details>

LGTM, thanks!

**Pull request merged**